src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -817,7 +817,7 @@ void emitter::emitInsSanityCheck(instrDesc* id)
             assert(isVectorRegister(id->idReg2()));
             assert(isVectorRegister(id->idReg3()));
             elemsize = optGetElemsize(id->idInsOpt());
-            assert(isValidVectorIndex(id->idOpSize(), elemsize, emitGetInsSC(id)));
+            assert(isValidVectorIndex(EA_16BYTE, elemsize, emitGetInsSC(id)));
```

Comment:
Was this changed to account for the simd8 result with simd16 selection?

Reply:
The size of the destination/source registers shouldn't matter here. For example, 
```asm
FMLA Vd.T, Vn.T, Vm.Ts[index]
```
index is **always** encoded as H:L (2 bits) when T is either 2S or 4S (and Ts is S).

In other words, the range of valid values for index is computed based on the assumption that Vm is 16 bytes.